### ISSUE TITLE
Passing a ruby class as a java.lang.Class into Java land #3454

### ISSUE THREAD

**Issue by carlosmrce** [jruby/jruby#3914]
I just tested thiss issue, and it seems the problem still exists! Was this supposed to be fixed on jruby-9.1.0.0 release?

**Comment by kares** [comment]
yes, and there's even a test-case that verifies [auto-reify works](https://github.com/jruby/jruby/commit/27d8098436d0d494a6882d2ae05c2b32682a8d76#diff-2a4dade2791e9bb085bbfbd439a2d9d6R52) ... what does not work, could you try reproducing maybe around that linked test-case ?

one (current) limitation is that it might not work when the original Ruby class structure changes after the class has been reified (so maybe you're running into that, otherwise expecting the original code to work).

**Comment by kares** [comment]
ping @carlosmrce ^^ ... current status - no understanding of what the issue is. please help us understand.
